File: ios/Classes/StreamHandlers/RemoteControlStreamHandler.swift
//
//  RemoteControlStreamHandler.swift
//  sdk_live_flutter_plugin
//
//  Created by Ross on 25.01.2022.
//

import Foundation
import Flutter
import ScreenMeetSDK

class RemoteControlStreamHandler: NSObject, FlutterStreamHandler {
    private let pm = SwiftChannelParamsManager()

    var sink: FlutterEventSink?
       
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
       
    func sendRemoteEvent(_ event: SMRemoteControlEvent) {
        guard let sink = sink else { return }
        sink(pm.remoteControlEventToDict(event))
    }
       
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/com/screenmeet/sdk_live_flutter_plugin/handlers/RemoteControlStreamHandler.kt
package com.screenmeet.sdk_live_flutter_plugin.handlers

import io.flutter.plugin.common.EventChannel

class RemoteControlStreamHandler: EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) { sink = events }
    override fun onCancel(arguments: Any?) { sink = null }

}